environment/environment.go

```diff
@@ -0,0 +1,924 @@
+	localURL string
+	// ClusterURL communicates with the dummy adapter from within the cluster
+	clusterURL string
+}
+
+// LocalURL is used for communication with the dummy adapter from outside the cluster
+func (ex *externalAdapter) LocalURL() string {
+	return ex.localURL
+}
+
+// ClusterURL is used for communication with the dummy adapter from within the cluster
+func (ex *externalAdapter) ClusterURL() string {
+	return ex.clusterURL
+}
+
+// SetVariable set the variable that's retrieved by the `/variable` call on the dummy adapter
+func (ex *externalAdapter) SetVariable(variable int) error {
+	_, err := http.Post(
+		fmt.Sprintf("%s/set_variable?var=%d", ex.localURL, variable),
+		"application/json",
+		nil,
+	)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// --------------- INTERNAL HELPERS ---------------
+// Convenience methods / functions
+
+func (env *environment) setEnvTools() error {
+	blockchainClient, err := client.NewBlockchainClient(env.network)
+	if err != nil {
+		return err
+	}
+	wallets, err := env.network.Wallets()
+	if err != nil {
+		return err
+	}
+	contractDeployer, err := contracts.NewContractDeployer(blockchainClient)
+	if err != nil {
+		return err
+	}
+	linkContract, err := contractDeployer.DeployLinkTokenContract(wallets.Default())
+	if err != nil {
+		return err
+	}
+	env.blockchainClient = blockchainClient
+	env.wallets = wallets
+	env.contractDeployer = contractDeployer
+	env.linkContract = linkContract
+	return err
+}
+
+// Deploys a hardhat pod to the cluster
+func (env *environment) deployHardhat() (string, error) {
+	hardhatDeploySpec, hardhatServiceSpec := newHardhat()
+	_, err := env.deploymentsClient.Create(context.Background(), hardhatDeploySpec, metav1.CreateOptions{})
+	if err != nil {
+		return "", err
+	}
+	hardhatService, err := env.servicesClient.Create(context.Background(), hardhatServiceSpec, metav1.CreateOptions{})
+	if err != nil {
+		return "", err
+	}
+	log.Info().Msg("Deployed Hardhat")
+	return "ws://" + hardhatService.Spec.ClusterIP + ":8545", err
+}
+
+// Builds all that's needed for launching a hardhat network for testing
+func newHardhat() (*appsv1.Deployment, *apiv1.Service) {
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "hardhat-network",
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": "hardhat-network"},
+			},
+			Template: apiv1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": "hardhat-network"},
+				},
+				Spec: apiv1.PodSpec{
+					Containers: []apiv1.Container{
+						{
+							Name:  "hardhat-network",
+							Image: "smartcontract/hardhat-network",
+							Ports: []apiv1.ContainerPort{
+								{
+									ContainerPort: 8545,
+								},
+							},
+							ReadinessProbe: &apiv1.Probe{
+								Handler: apiv1.Handler{
+									HTTPGet: &apiv1.HTTPGetAction{
+										Path: "/",
+										Port: intstr.FromInt(8545),
```

Comment:
Consider adding constants for often used ports/hosts/app names if it can help readability.